### DISCUSSION TITLE
Smartphone accelerometer data

### DISCUSSION THREAD

**Discussion by martinarielhartmann** [jspsych/jsPsych#2479] [Ideas]
Hi,

Accessing accelerometer data via from our phones seems to be possible with javascript:

- [https://codepen.io/ugs/pen/rmLQBB](https://codepen.io/ugs/pen/rmLQBB)
- [https://sensor-js.xyz/demo.html](https://sensor-js.xyz/demo.html)

I'm wondering whether someone has worked or is working on a plug in that can be used to collect this kind of data, assuming that it is possible to do so reliably. I'm especially interested in collecting 'movement to music' data, i.e. continuous movement data while playing a musical stimulus. Unfortunately, I don't speak much javascript. 

Thanks for your help!

**Comment by jodeleeuw** [comment]
This would be an ideal [extension](https://www.jspsych.org/7.1/overview/extensions/). The implementation would be pretty similar to the [mouse tracking extension](https://github.com/jspsych/jsPsych/blob/main/packages/extension-mouse-tracking/src/index.ts), which just records data based on mouse events. 

Happy to add this to a future todo list, but it would be a while before I can get to it.

**Comment by pasoneto** [comment]
Hi all, 

I just made a [pull request](https://github.com/jspsych/jspsych-contrib/pull/18) to the jspsych-contrib repo with the implementation of a device-motion extension. As @jodeleeuw suggested, this was based on the mouse tracking extension. You can test it [here](https://pasoneto.github.io/AccelerometerWeb/extensionCompiled/teste.html)

I wonder if my implementation could be added to the main repo and, if yes, how far away my code is from being accepted there.

Best,
